Add support for Rails 5.2+ Dirty API.

m272cels opened the pull request:
[Rails' Dirty API](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html) is updated in 5.2 and invalid usages of the old API must be changed in order to support newer versions of Rails.

A good summary of the new API is [here](https://www.fastruby.io/blog/rails/upgrades/active-record-5-1-api-changes).

m272cels committed:
Add support for Rails 5.2+ Dirty API.

phuibonhoa reviewed (approved):
If it’s not a ton of work, it would be worth adding specs for this one

m272cels committed:
Upgrade dummy app in specs to Rails 5.2

- Make update to dummy app Gemfile
- Comment out old deprecated configs
- Remove attr_accessible whitelisting in models
- Add manifest.js and secrets.yml
- Update gemspec to match

m272cels committed:
Fix logging for models and remove support for Rails <= 5.1

Previous logic would not log anything if has_changes_to_save? and
saved_changes? were both false, which could be the case if attempting
to log a destroyed record.

New logic ensures logging happens even if record does not change.

m272cels committed:
Replace update_attributes! with update! in specs.

update_attributes! is deprecated in Rails 6.1, in favor of update!.
update! is a better/shorter named alias for update_attributes!.

https://rubyinrails.com/2019/04/09/rails-6-1-activerecord-deprecates-update-attributes-methods/

m272cels commented:
I had a lot of trouble trying to get the specs running on the `spec/dummy` Rails 3.2 app, and since I was adding support for 5.2 anyway, I decided to:

1. Update `spec/dummy` to Rails 5.2 and update the specs to run against the updated app
2. Remove support for Rails <= 5.1 (older versions of Rails can be supported by older versions of Thoth)

m272cels committed:
Bump version to 2.0.0

phuibonhoa reviewed (approved):
Looks great, thanks for catching us up to modern times!
